fullstop-jobs/src/main/java/org/zalando/stups/fullstop/jobs/ami/FetchAmiJob.java

```diff
@@ -105,60 +107,75 @@ public void run() {
     private void runOn(final String account, final String region) {
         try {
             log.info("Scanning EC2 instances to fetch AMIs {}/{}", account, region);
-            final DescribeInstancesResult describeEC2Result = getDescribeEC2Result(account, region);
-            for (final Reservation reservation : describeEC2Result.getReservations()) {
-                for (final Instance instance : reservation.getInstances()) {
-                    if (violationService.violationExists(account, region, EVENT_ID, instance.getInstanceId(), OUTDATED_TAUPAGE)) {
-                        continue;
-                    }
-
-                    final Optional<Image> optionalImage = getAmiFromEC2Api(account, region, instance.getImageId());
-                    final Optional<Boolean> isTaupageAmi = optionalImage
-                            .filter(img -> img.getName().startsWith(taupageNamePrefix))
-                            .map(Image::getOwnerId)
-                            .map(taupageOwners::contains);
-
-                    // will not check for all non taupage ami
-                    // or images with taupage as name but created from another owner
-                    if (!isTaupageAmi.orElse(false)) {
-                        continue;
-                    }
+            final AmazonEC2Client ec2Client = clientProvider.getClient(
+                    AmazonEC2Client.class,
+                    account,
+                    getRegion(fromName(region)));
+            Optional<String> nextToken = empty();
+            do {
+                final DescribeInstancesRequest request = new DescribeInstancesRequest();
+                if (nextToken.isPresent()) {
+                    request.withNextToken(nextToken.get());
```

Comment:
The only difference here is, that `with*` returns the request for chaining calls. I know that it is not necessary in this line. However I used `withFilters` below, because a it has a nicer API (varargs) than `setFilters` (collection). And I thought it would be nice to use either `with*` or `set*` consistently.